community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/MultipleIndexPopulator.java

```diff
@@ -533,7 +539,18 @@ public boolean visit( NodeUpdates updates ) throws IndexPopulationFailedKernelEx
 
         private void add( NodeUpdates updates )
         {
-            forEachPopulation( population -> population.add( updates ) );
+            for ( IndexEntryUpdate indexUpdate : updates.forIndexes( populations.keySet(), NO_UNCHANGED_PROPERTIES ) )
+            {
+                IndexPopulation population = populations.get( indexUpdate.descriptor() );
```

Comment:
This seems to be the only reason why `populations` changed from a List to Map, I wonder if we couldn't somehow allow passing the `IndexPopulation` objects (opaquely) through the `NodeUpdates.forIndexes(...)` method, so that we can avoid this lookup altogether.

It seems to me that we add quite a bit of overhead to the data structures with very little gain. Although passing the `IndexPopulation` objects through might add code complexity instead. Feels like one of those things that we will not know until we try it out.